### ISSUE TITLE
test_ipc.py:TestListen.test_in_use marked as xfail

### ISSUE THREAD

**Issue by The-Compiler** [The-Compiler/qutebrowser#997]
This seems to fail on Archlinux but pass on Ubuntu/Debian, so we should investigate what's going on:

```
======================== FAILURES ========================
_________________ TestListen.test_in_use _________________

self = <test_ipc.TestListen object at 0x7f38226c1ba8>, qlocalserver = <PyQt5.QtNetwork.QLocalServer object at 0x7f3822657438>
ipc_server = <qutebrowser.misc.ipc.IPCServer object at 0x7f3822651948>, monkeypatch = <_pytest.monkeypatch.monkeypatch object at 0x7f3822bc2748>

    @pytest.mark.posix
    def test_in_use(self, qlocalserver, ipc_server, monkeypatch):
        monkeypatch.setattr('qutebrowser.misc.ipc.QLocalServer.removeServer',
                            lambda self: True)
        qlocalserver.listen('qute-test')
        with pytest.raises(ipc.AddressInUseError):  
>           ipc_server.listen()
E           Failed: DID NOT RAISE

tests/unit/misc/test_ipc.py:307: Failed
---------------------- Captured log ----------------------
ipc.py                     190 DEBUG    Calling setSocketOptions
ipc.py                     204 DEBUG    Listening as qute-test
```

It was marked as xfail in 3573b57130043aaa787e62a2d6b11a60d6d96ad0

**Comment by The-Compiler** [comment]
With this test script:

```python
from PyQt5.QtCore import QCoreApplication
from PyQt5.QtNetwork import QLocalServer

app = QCoreApplication([])

QLocalServer.removeServer('/tmp/ipc-test')

s1 = QLocalServer()
ok = s1.listen('/tmp/ipc-test')
print('s1: {}'.format(ok))
print(s1.errorString())

s2 = QLocalServer()
s2.setSocketOptions(QLocalServer.UserAccessOption)
ok = s2.listen('/tmp/ipc-test')
print('s2: {}'.format(ok))
print(s2.errorString())
print(s2.serverError())
```

with `setSocketOptions` we get:

```
s1: True
s2: True
-1
```

without `setSocketOptions` we get:

```
s1: True
s2: False
QLocalServer::listen: Address in use
8
```

This only happens on Archlinux because we don't use `setSocketOptions` on older Qt versions (see b135569d5c6e68c735ea83f42e4baf51f7972281).

**Comment by The-Compiler** [comment]
Reported upstream as [QTBUG-48635](https://bugreports.qt.io/browse/QTBUG-48635) - I'm guessing the best option right now is to not use `setSocketOptions` at all..
